src/librustc_mir/build/expr/as_rvalue.rs

```diff
@@ -181,6 +181,9 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
             ExprKind::Adt {
                 adt_def, variant_index, substs, fields, base
             } => { // see (*) above
+                let is_union = adt_def.adt_kind() == ty::AdtKind::Union;
+                let active_field_index = if is_union { Some(fields[0].name.index()) } else { None };
+
```

Comment:
It might be a better idea to use direct assignment here instead of an `Rvalue` - that would remove the need to change `AggregateKind::Adt` - alternatively, an `Union` `AggregateKind` could be added.